WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewController.swift

```diff
@@ -559,26 +559,38 @@ private extension ShippingLabelFormViewController {
 // MARK: - Top Banner
 //
 private extension ShippingLabelFormViewController {
-    /// Creates a Top Banner View containing the EU Shipping Notice.
+    /// Present a Top Banner View containing the EU Shipping Notice.
     ///
     func showTopBannerView() {
-        guard viewModel.shouldDisplayShippingNotice else {
-            return
+        viewModel.shouldDisplayEUShippingNotice { [weak self] shouldDisplay in
+            guard let self = self, shouldDisplay else {
+                return
+            }
+
+            let topBannerView = self.createEUShippingNoticeBannerView()
+            self.topBannerView = topBannerView
+            let headerContainer = UIView(frame: CGRect(x: 0, y: 0, width: Int(self.tableView.frame.width), height: Int(Constants.headerDefaultHeight)))
+            headerContainer.addSubview(topBannerView)
+            headerContainer.pinSubviewToAllEdges(topBannerView, insets: Constants.headerContainerInsets)
+            self.tableView.tableHeaderView = headerContainer
+            self.tableView.updateHeaderHeight()
         }
+    }
 
-        let topBannerView = EUShippingNoticeTopBannerFactory.createTopBanner(
+    /// Creates the Shipping Notice Top banner with the appropriate actions.
+    ///
+    func createEUShippingNoticeBannerView() -> TopBannerView {
+        EUShippingNoticeTopBannerFactory.createTopBanner(
             onDismissPressed: {
-                self.hideTopBannerView()
+                self.viewModel.setEUShippingNoticeDismissState(isDismissed: true) { success in
+                    if success {
+                        self.hideTopBannerView()
+                    }
```

Comment:
I mentioned this on the previous PR, but just mentioning it again since this code is being changed here: It's safer to capture `self` weakly in these closures.

Reply:
Thanks for the reminder, Rachel! Also fixed in https://github.com/woocommerce/woocommerce-ios/commit/16f2752215bbf475b4e9dcf3cf9882eb1ff69540.